### PR TITLE
Also test GetSortedCallstackReportFromFunctionAddresses

### PR DESCRIPTION
**Pull request by dpallotti**

### CODE REVIEW

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/ClientModel/SamplingDataPostProcessorTest.cpp [hunk 1]
```diff
@@ -71,6 +73,49 @@ MATCHER_P(ThreadSampleDataEq, that, "") {
...
+  std::sort(rhs_resorted_callstacks_counts.begin(), rhs_resorted_callstacks_counts.end(),
+            callstack_count_less);
+
+  auto callstack_count_equal = [](const CallstackCount& lhs, const CallstackCount& rhs) {
+    return lhs.count == rhs.count && lhs.callstack_id == rhs.callstack_id;
+  };
+
+  return lhs.total_callstack_count == rhs.total_callstack_count &&
+         std::equal(lhs_resorted_callstacks_counts.begin(), lhs_resorted_callstacks_counts.end(),
+                    rhs_resorted_callstacks_counts.begin(), callstack_count_equal);
```
[the remark this answers is not in the document]
Comment (dpallotti) [reply src/ClientModel/SamplingDataPostProcessorTest.cpp]: Aaah I read the documentation of `std::equal` wrong. Changed the two `std::equal` to this version.

File: src/ClientModel/SamplingDataPostProcessorTest.cpp [hunk 2]
```diff
@@ -220,10 +266,12 @@ class SamplingDataPostProcessorTest : public ::testing::Test {
     // D  = kFunction4Instruction1AbsoluteAddress.
     // There are the CallstackEvents that are added, innermost frame at the top. Note that:
     //   - the first and second CallstackEvents have the same Callstack;
-    //   - the last CallstackEvent has two identical frames.
+    //   - the last CallstackEvent has two identical frames;
+    //   - the number under each CallstackEvent is the callstack id.
     // C    C    D    C'   C
     // B    B    C    C    C
     // A    A    A    A    A
```
[the remark this answers is not in the document]
Comment (dpallotti) [reply src/ClientModel/SamplingDataPostProcessorTest.cpp:275]: It's not really connected to the previous comment as it tests something else, but you're right. I'll look into this now, I'll probably add a commit.

[the remark this answers is not in the document]
Comment (dpallotti) [reply src/ClientModel/SamplingDataPostProcessorTest.cpp]: Done, changed the A C D one to A B C D.

**Pull request merged**